jwql/website/apps/jwql/templates/engineering_database.html

```diff
@@ -0,0 +1,130 @@
+                                    <strong>{{ error|escape }}</strong>
+                                </div>
+                            {% endfor %}
+                        {% endfor %}
+                    {% endif %}
+
+                    <!--Django Cross-Site Request Forgery magic-->
+                    {{ csrf_input }}
+
+                    <!--Show the field forms-->
+                    {% for field in mnemonic_name_search_form %}
+                        <div class="homepage_form_fieldWrapper">
+                            {{ field }}
+                            {% if field.help_text %}
+                                <p class="help">{{ field.help_text|safe }}</p>
+                            {% endif %}
+                        </div>
+                    {% endfor %}
+                    <button name=mnemonic_name_search class="btn btn-primary" type="submit"><span class="fas fa-search"></span></button>
+
+                </form>
+            </div>
+            <div class="mnemonic_name_search_col">
+                <b>Description and details of menemonic from EDB:</b>
+                <p></p>
+                <table>
+                <ul>
+                {% for key, value in mnemonic_result.items() %}
+                    <tr> <td><i>{{ key }}</i> :<td> <td>{{ value }}<td></tr>
+                {% endfor %}
+                </ul>
+                </table>
+            </div>
+        </div>
+
+        <p></p>
+
+        <h4>Query for records of an EDB mnemonic</h4>
+
+        <div class="mnemonic_query_section">
+
+            <b>Submit a mnemonic identifier with a time range to view the corresponding EDB records.</b>
+
+            <p></p>
+            <!--Load the EDB mnemonic search form from the view-->
+            <form action="" method="post" id="mnemonic_name_search">
+                <!--Show any errors from a previous form submission-->
+                {% if mnemonic_query_form.errors %}
+                    {% for field in mnemonic_query_form %}
+                        {% for error in field.errors %}
+                            <div class="alert alert-danger">
+                                <strong>{{ error|escape }}</strong>
+                            </div>
+                        {% endfor %}
+                    {% endfor %}
+                {% endif %}
+
+                <!--Django Cross-Site Request Forgery magic-->
+                {{ csrf_input }}
+
+                <!--Show the field forms-->
+                {% for field in mnemonic_query_form %}
+                    <div class="mnemonic_query_field">
+                        {{ field }}
+                        {% if field.help_text %}
+                            <p class="help">{{ field.help_text|safe }}</p>
+                        {% endif %}
+                    </div>
+                {% endfor %}
+                <button name=mnemonic_query class="btn btn-primary" type="submit"><span class="fas fa-search"></span></button>
```

Comment:
Weirdly, I can succesfully return the description and details of a mnemonic with the search form above, and I can query data using in the Jupyter notebook you wrote. However, when I try to query for records here, I get this error: 

<img width="1452" alt="screen shot 2019-02-05 at 4 12 43 pm" src="https://user-images.githubusercontent.com/12650803/52304644-34728e00-2961-11e9-9810-66e4a405ec11.png">

I think that I should have all my tokens and permissions in order (since I can get data using the notebook), and I tried two different mnemonics with the same result. Any idea why this error might be happening?

Reply:
This is weird. I have never encountered this, but I have sometimes been in a situation where I had to restart the app to get rid of some 'history' that was messing up things. It may also be that the EDB was simply down? 
Can you maybe try again after a restart?

Reply:
@Johannes-Sahlmann @laurenmarietta I dug into this a bit.  I notice that when I put 

```python
    json_data = result[0].json()
    print(json_data['status'], json_data['msg'])
```

At the beginning of the `process_mast_service_request_result()` function, and I query for records of an EDB mnemonic on the webpage, two separate queries happen:

```
COMPLETE
ERROR Must be logged in as an authorized user to access this service.
```

The first one seems to be successful and return results (can confirm that when I print the full JSON data I get an absolute flood of data) and the second one indicates that I am not logged in.

Reply:
Oh! Yup, it is working now after restarting the server. Thanks for the tip - hopefully others don't run into whatever problem I was having.

Reply:
@Johannes-Sahlmann Upon further digging, I realize that the successful query is happening in `mnemonic_inventory()` which uses the `mast_edb_mnemonic_service`, while the unsuccessful query is happening in `query_single_mnemonic` which uses the `mast_edb_timeseries_service`.  Perhaps it could it be that each service has its own set of authorized users?

Reply:
So if you look at the console output when running the query from the web app, you see the successful login using the mast_token `INFO: MAST API token accepted, welcome ...`?

Could it be that you are already authenticated through the login and that there is interference between mast_token and authentication?

Reply:
@Johannes-Sahlmann Yes, I do see `INFO: MAST API token accepted, welcome ...` when I first start the web server (before doing anything else).  I also tried forcing a log in again in the code within `query_single_mnemonic` -- I again see the acceptance message but it does not change any behavior.

I notice that if I remove my token altogether, or use some invalid dummy token, I still am able to run the `mast_edb_mnemonic_service` successfully still despite not being logged in.  Perhaps the token I am using is not authenticated for anything. 

I was authorized by Deborah Kenny yesterday and can confirm that I can run queries successfully through the MAST EDB portal.

Did you have to do anything special to generate and use your `mast_token`?

Reply:
There is nothing special in my way of creating the `mast_token` I think.